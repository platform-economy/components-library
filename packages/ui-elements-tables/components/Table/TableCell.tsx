import styled from 'styled-components';

export type TableCellProps = {
  align?: 'left' | 'center' | 'right';
};

const StyledTableCell = styled.td<TableCellProps>`
  transition: padding .5s;
  padding: 18px 16px;
  text-align: ${({ align }) => align || 'initial'};

  &:first-child {
    padding-left: 30px;
  }

  &:last-child {
    padding-right: 30px;
  }
`;

export const TableCell = StyledTableCell;
