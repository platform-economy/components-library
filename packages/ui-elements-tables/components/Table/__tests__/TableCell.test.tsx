import React from 'react';
import * as TestRenderer from 'react-test-renderer';
import 'jest-styled-components';

// Component
import { TableCell } from '../TableCell';

describe('ui-elements-tables', () => {
  describe('Components', () => {
    describe('TableCell', () => {
      it('Should render without throwing an exception', () => {
        const tree = TestRenderer
          .create(<TableCell />)
          .toJSON();

        expect(tree).toMatchSnapshot();
      });
    });
  });
});
