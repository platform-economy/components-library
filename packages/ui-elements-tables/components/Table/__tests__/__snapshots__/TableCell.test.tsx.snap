// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ui-elements-tables Components TableCell Should render without throwing an exception 1`] = `
.c0 {
  -webkit-transition: padding .5s;
  transition: padding .5s;
  padding: 18px 16px;
  text-align: initial;
}

.c0:first-child {
  padding-left: 30px;
}

.c0:last-child {
  padding-right: 30px;
}

<td
  className="c0"
/>
`;
